src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -562,6 +564,21 @@ private void UpdateCallsiteExecutionState(Object callsiteData, RuntimeCore runti
                 }
             }*/
         }
+        //TODO write a unit test for this.
+        private static bool CheckIfTraceDataIsLegacySOAPFormat(string base64EncodedTraceData)
+        {
+            var data = Convert.FromBase64String(base64EncodedTraceData);
+            if (data.Length > 71)
+            {
+                var header = Encoding.UTF8.GetString(data, 0, 72);
+                if (header == @"<SOAP-ENV:Envelope xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance""")
```

Comment:
Is it always 2001?

Reply:
in all test data I saw - we could probably make the header even shorter / more tolerant.

Reply:
Mayabe just check for `<SOAP-ENV:Envelope`..